cloudinit/util.py

```diff
@@ -2894,13 +2894,13 @@ def get_proc_ppid(pid):
     ppid = 0
     try:
         contents = load_file("/proc/%s/stat" % pid, quiet=True)
+        if contents:
+            # see proc.5 for format
+            m = re.search(r"^\d+ \(.+\) [RSDZTtWXxKWP] (\d+)", str(contents))
```

Comment:
Looks like the man page is incomplete as I see an `I` state on my kinetic system. Als you have a duplicate `W` we can drop.

```suggestion
            m = re.search(r"^\d+ \(.+\) [RSDZTtWXxKPI] (\d+)", str(contents))```
Testing on my system:

Reply:
Nice catch :)

Looks like that was introduced in https://github.com/torvalds/linux/commit/06eb61844d841d0032a9950ce7f8e783ee49c0d0, and this hasn't made its way into the upstream man pages project yet. I just submitted a [patch](https://lore.kernel.org/linux-man/20220805044812.glotze62j2dldrdw@isa/T/#u) to add it.

I don't see any states in the array that aren't in our list (besides the one you found), so I think that covers our bases on the linux front (no clue about the BSDs).

Reply:
Thanks for the due diligence to check on other states in the kernel and finding the source related to that diff and fixing upstream for it :) All good here!

Reply:
ahh right and BSD doesn't surface this info in `/proc` in many cases due to /proc deprecation I believe.